src/protocol/sort/shuffle.rs

```diff
@@ -103,70 +119,121 @@ impl<'a, F: Field, S: Step + SpaceIndex> Shuffle<'a, F, S> {
         try_join_all(reshares).await
     }
 
-    /// `single_shuffle` is called for the helpers
-    /// i)   2 helpers generate random sequence using shared prss random number
+    /// `single` is called for the helpers
+    /// i)   2 helpers receive permutation pair and choose the permutation to be applied
     /// ii)  2 helpers apply the permutation to their shares
     /// iii) reshare to `to_helper`
-    #[allow(clippy::cast_possible_truncation)]
-    async fn single_shuffle<M: Mesh, G: Gateway<M, S>>(
+    async fn single<M: Mesh, G: Gateway<M, S>>(
```

Comment:
select a different name. Perhaps: "shuffle_or_unshuffle_once"

Reply:
thanks for the suggestion!